crates/js-sys/src/lib.rs

```diff
@@ -188,6 +188,23 @@ extern "C" {
     #[wasm_bindgen(method, js_name = findIndex)]
     pub fn find_index(this: &Array, predicate: &mut dyn FnMut(JsValue, u32, Array) -> bool) -> i32;
 
+    /// The flat() method creates a new array with all sub-array elements concatenated into it
+    /// recursively up to the specified depth.
+    ///
+    /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat)
+    #[wasm_bindgen(method)]
+    pub fn flat(this: &Array, depth: i32) -> Array;
+
+    /// The flatMap() method first maps each element using a mapping function, then flattens
+    /// the result into a new array.
+    ///
+    /// [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)
+    #[wasm_bindgen(method, js_name = flatMap)]
+    pub fn flat_map(
+        this: &Array,
+        callback: &mut dyn FnMut(JsValue, u32, Array) -> Vec<JsValue>,
```

Comment:
I wasn't sure of the best type to use for the callback here. In JS, it's possible for the `flatMap` callback to return a non-array value `v`, which has the same effect as returning the one-element array `[v]`. But it seems cleaner to be more strict in the Rust type.

I think `Vec<JsValue>` is the most appropriate return value here (as opposed to e.g. a slice), but it might be worth double-checking this; I don't have a lot of experience with lifetimes.